### DISCUSSION TITLE
How to update a transitive dependency with `deno install`?

### DISCUSSION THREAD

**Discussion by nicolo-ribaudo** [denoland/deno#26662] [Q&A]
At https://github.com/nicolo-ribaudo/local-highlight-registry/tree/efd1ff4fdc2091a7fc5bc6901743c2461037573b, I had to add `"rollup": "npm:rollup@4.24.0"` to my `deno.jsonc` because I am using Vite, and `rollup@4.24.1` had a bug that made it impossible to install it with Deno.

That Rollup bug has been fixed, and now I would like to update Rollup to the fixed version. I don't want to simply delete the lockfile, because I don't want to update other unrelated dependencies.

How do I do it?

I tried:
- deleting `"rollup": "npm:rollup@4.24.0"` from `deno.jsonc` and running `deno install`. This has no effect.
- running `deno install npm:rollup`. This replaced `npm:rollup@4.24.0` with `npm:rollup@^4.24.3` in my `deno.jsonc` file. However, they are _both_ present in the lockfile and the Vite entry in the lockfile has been updated to explicilty use the old version instead of the new one
- running `deno install npm:rollup` and `deno uninstall npm:rollup`. Same result as above, except that (as expected) I don't have Rollup anymore in my `deno.jsonc` file
- manually removing the rollup entry from `deno.jsonc` and from `deno.lock`'s `npm` object and running `deno install`. It fails with:
  ```
  error: Failed reading lockfile at '/Users/nic/Documents/dev/github.com/nicolo-ribaudo/local-highlight-registry/deno.lock'
  
  Caused by:
    0: Failed deserializing. Lockfile may be corrupt
    1: npm package 'rollup' was not found and could not have its version resolved
  ```

**Comment by bartlomieju** [comment]
I think there might be no way besides removing the lock file are the moment. We got a PR cooking that would allow to do that (https://github.com/denoland/deno/pull/26219) that we want to ship in Deno v2.1.

**Reply by nicolo-ribaudo** [reply]
Ok thank you, I'll wait :)

**Reply by lukehsiao** [reply]
The referenced PR became https://github.com/denoland/deno/pull/26942, and still does NOT support updated transitive dependencies :(
